app/src/main/kotlin/com/wire/android/ui/home/conversations/messages/ConversationMessagesViewModel.kt

```diff
@@ -276,6 +282,32 @@ class ConversationMessagesViewModel @Inject constructor(
         }
     }
 
+    fun updateImageOnFullscreenMode(message: UIMessage?) {
+        imageMessageOnFullscreen = message
+    }
+
+    fun checkPendingActions(onMessageReply: (UIMessage) -> Unit) = viewModelScope.launch {
+        savedStateHandle.getBackNavArg<Pair<String, String>>(EXTRA_ON_MESSAGE_REACTED)?.let { (messageId, emoji) ->
+            toggleReaction(messageId, emoji)
+        }
+        savedStateHandle.getBackNavArg<String>(EXTRA_ON_MESSAGE_REPLIED)?.let { messageId ->
+            imageMessageOnFullscreen?.let { onFullscreenMessage ->
+                updateImageOnFullscreenMode(null) // We need to reset the imageOnFullscreenMode as we handle it here
+                if (onFullscreenMessage.messageHeader.messageId == messageId) {
```

Comment:
what is being checked here ? maybe some local variable would help follow what is happening ?

Reply:
This is just a safety check that should always succeed. The `messageId` variable stored in the `SavedStateHandle` from the navigation manager should match with the messageId of the image that we replied to in the `MediaGalleryScreen` 
I can try to add some documentation for others to understand it better.